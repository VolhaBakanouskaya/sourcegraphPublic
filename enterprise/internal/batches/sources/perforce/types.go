package perforce

import "github.com/sourcegraph/sourcegraph/internal/extsvc/perforce"

// from @varsanojidan, in a comment on the PR:
// You only really need this if the Changelist model doesn't contain information about changelist builds or reviews
// (if perforce has such a concept), if you can get all the info from Changelist directly,
// then you can just use the Changelist type instead.
//
// at some point we will probably want to add info about CLs that we would get from Swarm or elsewhere
// so I'll leave this in here
type AnnotatedChangelist struct {
	*perforce.Changelist
}
