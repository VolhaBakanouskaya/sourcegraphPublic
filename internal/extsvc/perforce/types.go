package perforce

import (
	"time"
)

// TODO: this was just blindly copied from the ADO implementation in order to make things "work" initially
// need to probably switch most of this out for Changelist vernacular

type ChangelistStatus string

type CreatorInfo struct {
	ID          string `json:"id"`
	DisplayName string `json:"displayName"`
	UniqueName  string `json:"uniqueName"`
	URL         string `json:"url"`
	ImageURL    string `json:"imageUrl"`
}

type Changelist struct {
	Depot        Depot            `json:"depot"`
	ID           int              `json:"changelistId"`
	Status       ChangelistStatus `json:"status"` // pending, submitted, shelved
	CreationDate time.Time        `json:"creationDate"`
	CreatedBy    CreatorInfo      `json:"createdBy"`
	Title        string           `json:"title"`
	Description  string           `json:"description"`
	URL          string           `json:"url"`
}

type Repository struct {
	ID         string `json:"id"`
	Name       string `json:"name"`
	CloneURL   string `json:"remoteURL"`
	IsDisabled bool   `json:"isDisabled"`
}
