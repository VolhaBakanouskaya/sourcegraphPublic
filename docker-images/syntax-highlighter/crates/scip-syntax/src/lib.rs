use anyhow::Result;
use scip::types::Occurrence;
use scip_treesitter_languages::parsers::BundledParser;

pub mod ctags;
pub mod globals;
pub mod languages;
pub mod locals;
pub mod ts_scip;

pub fn get_globals(parser: BundledParser, source_bytes: &[u8]) -> Option<Result<Vec<Occurrence>>> {
    let mut config = languages::get_tag_configuration(parser)?;
    let tree = config.parser.parse(source_bytes, None).unwrap();
    Some(globals::parse_tree(&mut config, &tree, source_bytes))
}

pub fn get_locals(parser: BundledParser, source_bytes: &[u8]) -> Option<Result<Vec<Occurrence>>> {
    let mut config = languages::get_local_configuration(parser)?;
    let tree = config.parser.parse(source_bytes, None).unwrap();
    Some(locals::parse_tree(&mut config, &tree, source_bytes))
}
