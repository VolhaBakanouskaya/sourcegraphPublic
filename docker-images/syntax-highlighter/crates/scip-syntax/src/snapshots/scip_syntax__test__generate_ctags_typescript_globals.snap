---
source: crates/scip-syntax/src/lib.rs
expression: "String::from_utf8_lossy(buf_writer.buffer())"
---
{"_type":"tag","name":"MyEnum","path":"globals.ts","language":"go","line":16,"kind":"type","scope":null}
{"_type":"tag","name":"go","path":"globals.ts","language":"go","line":19,"kind":"variable","scope":"MyEnum"}
{"_type":"tag","name":"rust","path":"globals.ts","language":"go","line":18,"kind":"variable","scope":"MyEnum"}
{"_type":"tag","name":"zig","path":"globals.ts","language":"go","line":17,"kind":"variable","scope":"MyEnum"}
{"_type":"tag","name":"MyInterface","path":"globals.ts","language":"go","line":11,"kind":"type","scope":null}
{"_type":"tag","name":"sayBruh","path":"globals.ts","language":"go","line":13,"kind":"method","scope":"MyInterface"}
{"_type":"tag","name":"bruh","path":"globals.ts","language":"go","line":12,"kind":"variable","scope":"MyInterface"}
{"_type":"tag","name":"MyClass","path":"globals.ts","language":"go","line":1,"kind":"type","scope":null}
{"_type":"tag","name":"also_private_method","path":"globals.ts","language":"go","line":8,"kind":"method","scope":"MyClass"}
{"_type":"tag","name":"#private_method","path":"globals.ts","language":"go","line":7,"kind":"method","scope":"MyClass"}
{"_type":"tag","name":"public_method","path":"globals.ts","language":"go","line":6,"kind":"method","scope":"MyClass"}
{"_type":"tag","name":"also_private_field","path":"globals.ts","language":"go","line":4,"kind":"variable","scope":"MyClass"}
{"_type":"tag","name":"#private_field","path":"globals.ts","language":"go","line":3,"kind":"variable","scope":"MyClass"}
{"_type":"tag","name":"public_field","path":"globals.ts","language":"go","line":2,"kind":"variable","scope":"MyClass"}
{"_type":"tag","name":"func","path":"globals.ts","language":"go","line":25,"kind":"method","scope":null}
{"_type":"tag","name":"global2","path":"globals.ts","language":"go","line":23,"kind":"variable","scope":null}
{"_type":"tag","name":"global1","path":"globals.ts","language":"go","line":22,"kind":"variable","scope":null}

